### PR TITLE
Add options prop in dashboard menu item

### PR DESCRIPTION
**Pull request by bodo22**
To customize the DashboardMenuItem component, e.g. the MenuItem's `style` prop, the `options` prop is spread at the end of DashboardMenuItem.

### CODE REVIEW

File: packages/react-admin/src/mui/layout/DashboardMenuItem.js [hunk 1]
```diff
@@ -16,20 +16,22 @@ const styles = {
     },
 };
 
-const DashboardMenuItem = ({ classes, onClick, translate }) => (
+const DashboardMenuItem = ({ classes, onClick, translate, options }) => (
```
Comment (djhi) [inline packages/react-admin/src/mui/layout/DashboardMenuItem.js]: why not just `...props` ?

Comment (bodo22) [reply packages/react-admin/src/mui/layout/DashboardMenuItem.js]: I followed naming in most input components but now realized the most remaining prop name is props :)

File: packages/react-admin/src/mui/layout/DashboardMenuItem.js [hunk 2]
```diff
@@ -16,20 +16,22 @@ const styles = {
     },
 };
 
-const DashboardMenuItem = ({ classes, onClick, translate }) => (
+const DashboardMenuItem = ({ classes, onClick, translate, props }) => (
```
Comment (djhi) [inline packages/react-admin/src/mui/layout/DashboardMenuItem.js]: Not what I meant. Why don't you simply pass all extra props ?

```js
const DashboardMenuItem = ({ classes, onClick, translate, ...props }) => (
```

Comment (bodo22) [reply packages/react-admin/src/mui/layout/DashboardMenuItem.js]: I was following the pattern used in most input components, e.g. `BooleanInput`, `TextInput`. Should I still change it?

Comment (djhi) [reply packages/react-admin/src/mui/layout/DashboardMenuItem.js]: Yes please, we're going to apply this to all components anyway for `next`

Comment (djhi) [reply packages/react-admin/src/mui/layout/DashboardMenuItem.js]: Besides, sorry but you'll have to rebase as we updated `next` to add `className` to all components too

**Comment by djhi** [comment]
The commits in this PR have been messed up. Can you either clean them or make a new PR ?

**Pull request closed without merging**